src/kontrol/prove.py

```diff
@@ -680,20 +720,37 @@ def _update_cterm_from_node(cterm: CTerm, node: KCFG.Node, contract_name: str) -
     singlecall_cell = node.cterm.cell('SINGLECALL_CELL')
     gas_cell = node.cterm.cell('GAS_CELL')
     callgas_cell = node.cterm.cell('CALLGAS_CELL')
-    new_accounts = [CTerm(account, []) for account in flatten_label('_AccountCellMap_', new_accounts_cell)]
+
+    all_accounts = flatten_label('_AccountCellMap_', new_accounts_cell)
+
+    new_accounts = [CTerm(account, []) for account in all_accounts if (type(account) is KApply and account.is_cell)]
+    non_cell_accounts = [account for account in all_accounts if not (type(account) is KApply and account.is_cell)]
+
     new_accounts_map = {account.cell('ACCTID_CELL'): account for account in new_accounts}
-    test_contract_account = new_accounts_map[Foundry.address_TEST_CONTRACT()]
-
-    new_accounts_map[Foundry.address_TEST_CONTRACT()] = CTerm(
-        set_cell(
-            test_contract_account.config,
-            'CODE_CELL',
-            KEVM.bin_runtime(KApply(f'contract_{contract_name}')),
-        ),
-        [],
-    )
 
-    new_accounts_cell = KEVM.accounts([account.config for account in new_accounts_map.values()])
+    if Foundry.address_TEST_CONTRACT() in new_accounts_map:
+        test_contract_account = new_accounts_map[Foundry.address_TEST_CONTRACT()]
+
+        new_accounts_map[Foundry.address_TEST_CONTRACT()] = CTerm(
+            set_cell(
+                test_contract_account.config,
+                'CODE_CELL',
+                KEVM.bin_runtime(KApply(f'contract_{contract_name}')),
+            ),
+            [],
+        )
+    if config_type == ConfigType.SUMMARY_CONFIG:
+        for account_id, account in new_accounts_map.items():
+            new_accounts_map[account_id] = CTerm(
+                set_cell(
+                    account.config,
+                    'STORAGE_CELL',
+                    abstract_term_safely(account.cell('STORAGE_CELL'), base_name='STORAGE', sort=KSort('Map')),
+                ),
+                [],
+            )
```

Comment:
Are there other aspects of the state that can be set up in the constructor but potentially modified by later function calls that we also need to abstract?

Reply:
Since the rules of the CSE-related outputs haven't changed, it would seem that everything is getting abstracted correctly. We should keep an eye on this.